### PR TITLE
Ben footer

### PR DESCRIPTION
**Pull request by beneke090**

### CODE REVIEW

File: src/components/Footer/Footer.js [hunk 1]
```diff
@@ -42,62 +102,109 @@ const Footer = ({
...
+                <FaInfo /> {t(isOpen)}
+              </div>
+            }
+            classNameChildren="tm-dialog-content"
+            onClose={() => setIsOpen(null)}
+            cancelDraggable=".tm-dialog-content"
+            isOpen={isOpen}
+            isModal
+          >
+            <LegalLines
```
Comment (edellucien) [inline src/components/Footer/Footer.js]: I wouldn't call this component 'LegalLines' but rather some generic name, such as 'FooterDialogContent'

File: src/components/Footer/Footer.js [hunk 2]
```diff
@@ -42,62 +102,109 @@ const Footer = ({
...
-          dispatchSetLanguage(opt.value);
-        }}
-      />
-    </div>
-  </RSFooter>
-);
+}) => {
+  const [isOpen, setIsOpen] = useState(null);
+
+  // const getDialogContent = value => {
```
Comment (edellucien) [inline src/components/Footer/Footer.js]: Can removed commented parts

File: public/lang/locale-de-main.json [hunk 3]
```diff
@@ -104,6 +104,7 @@
   "ch.sbb.zweitausbildung": "Ausbildungen Personenverkehr SBB",
   "ch.sbb.zweitausbildung-desc": "Die Übersichtskarte Geografie wird in der Ausbildung des Zugpersonals SBB, des Schalterpersonals SBB und von Mitarbeitenden im Contact Center Brig verwendet. Mit Hilfe der Karte lernen die Auszubildenden, definierte Ortschaften, Hauptlinien, touristische Reiseziele und touristische Strecken geografisch zuzuordnen.<p>Verantwortlich:</p>HR-BIL-SKK-PM (VM und VS), <a href=\"mailto:pm.skk@sbb.ch\">pm.skk@sbb.ch</a></p>",
   "connection_plural": "Verbindung",
+  "contact_content": "<p>Haben Sie Fragen, Verbesserungsvorschläge, Hinweise auf Unstimmigkeiten oder Interesse an Karten und Plänen? Dann nehmen Sie bitte mit uns Kontakt auf!</p><p><b>SBB AG</b><br />Infrastruktur – Anlagen und Technologie – Operation Center Technik – Technical Management<br />Fachbus Trafimage<br />Poststrasse 6<br />3072 Ostermundingen<br />Schweiz</p><p><a href=\"http://www.trafimage.ch\" tabindex=\"2\">www.trafimage.ch</a><br /><a href=\"mailto:trafimage@sbb.ch\" tabindex=\"2\">trafimage@sbb.ch</a><br /></p>",
```
Comment (edellucien) [inline public/lang/locale-de-main.json]: unused {Key: value} can be removed

File: src/components/Dialog/Dialog.scss [hunk 4]
```diff
@@ -5,11 +5,16 @@
...
     overflow: hidden;
+    font-size: 14px;
+    font-family: Arial, sans-serif;
 
     .tm-dialog-header {
       padding: 0 20px;
       text-align: left;
       font-weight: bold;
+      border-bottom: 1px solid #e5e5e5;
+      min-height: 16.42857143px;
```
Comment (edellucien) [inline src/components/Dialog/Dialog.scss]: You can maybe define a height: auto, or a defined value instead

File: src/components/Footer/Footer.js [hunk 5]
```diff
@@ -34,6 +38,62 @@ const numberFormat = coords => {
 
   return coordStr;
 };
+/*
```
Comment (edellucien) [inline src/components/Footer/Footer.js]: Commented part can be removed

File: src/components/Footer/Footer.js [hunk 6]
```diff
@@ -42,62 +102,109 @@ const Footer = ({
...
+
+          <Dialog
+            title={
+              <div>
+                <FaInfo /> {t(isOpen)}
+              </div>
+            }
+            classNameChildren="tm-dialog-content"
+            onClose={() => setIsOpen(null)}
+            cancelDraggable=".tm-dialog-content"
```
Comment (edellucien) [inline src/components/Footer/Footer.js]: This component is not draggable, so you don't need this prop

File: src/components/legals/LegalLines.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  impressum,
+};
+
+const propTypes = {
+  language: PropTypes.string,
+  doc: PropTypes.oneOf(Object.keys(docs)),
+};
+
+const defaultProps = {
+  doc: Object.keys(docs)[0],
```
Comment (edellucien) [inline src/components/legals/LegalLines.js]: I would call this prop 'body' instead

File: src/components/legals/LegalLines.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  language: 'de',
+};
+
+const LegalLines = ({ language, doc }) => {
+  if (!Object.keys(docs).includes(doc) || !docs[doc][language]) {
+    return <Redirect to="/bern" />;
```
Comment (edellucien) [inline src/components/legals/LegalLines.js]: This Redirect was relevant in the example i gave you, but isn't here, you can remove it.
The props 'doc' & 'language' supposedly always leads to a doc, else return null

Comment (beneke090) [reply src/components/legals/LegalLines.js]: Had to change the *.test.js, but success

File: src/components/legals/LegalLines.js (hunk 7 above)
Comment (beneke090) [reply src/components/legals/LegalLines.js]: changed

File: src/components/Footer/Footer.js [hunk 9]
```diff
@@ -42,62 +46,95 @@ const Footer = ({
...
-}) => (
-  <RSFooter className="wkp-footer">
-    <div className="wkp-footer-left">
-      <Copyright
-        layerService={layerService}
-        format={f => `${t('Geodaten')} ${f}`}
-      />
-    </div>
+}) => {
+  const [isOpen, setIsOpen] = useState(null);
```
Comment (oterral) [inline src/components/Footer/Footer.js]: Usually the setter of a isXXX  variable is called `setOpen`  .

Comment (oterral) [reply src/components/Footer/Footer.js]: Usally a isXXXX variable means it's a boolean , which is not the case here, call it just `open` or  `dialogOpen` whatever

File: src/components/legals/FooterDialogContent.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
+import PropTypes from 'prop-types';
+
+import KontaktDE from './kontakt/KontaktDE';
+import KontaktFR from './kontakt/KontaktFR';
+import KontaktIT from './kontakt/KontaktIT';
+import KontaktEN from './kontakt/KontaktEN';
```
Comment (oterral) [inline src/components/legals/FooterDialogContent.js]: pls use english

File: src/components/legals/FooterDialogContent.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+import LegalDE from './rechtliches/LegalDE';
+import LegalFR from './rechtliches/LegalFR';
+import LegalIT from './rechtliches/LegalIT';
+import LegalEN from './rechtliches/LegalEN';
```
Comment (oterral) [inline src/components/legals/FooterDialogContent.js]: pls use english

File: src/components/Footer/Footer.js (hunk 1 above)
Comment (oterral) [reply src/components/Footer/Footer.js]: LegalLines was generic. FooterDialogContent is not.  Pls go back to the old name.

File: src/components/legals/FooterDialogContent.js (hunk 11 above)
Comment (oterral) [reply src/components/legals/FooterDialogContent.js]: Call the folder `Legal` like the compoenents

File: src/components/Dialog/Dialog.scss [hunk 12]
```diff
@@ -1,15 +1,24 @@
...
+
 .tm-dialog,
 .tm-modal {
   z-index: 10000;
 
   .tm-dialog-container {
-    width: 320px;
-    overflow: hidden;
+    font-size: 14px;
+    font-family: Arial, sans-serif;
```
Comment (oterral) [inline src/components/Dialog/Dialog.scss]: This must be already defined higher, I think.

**Comment by oterral** [comment]
New PR https://github.com/geops/trafimage-maps/pull/77

**Pull request closed without merging**